content/docs/reference/types/reference.mdx

```diff
@@ -140,3 +140,113 @@ You can search your reference with reference selector
+```ts
+const referenceField = {
+  label: 'Author',
+  name: 'author',
+  type: 'reference',
+  ui: {
+    selectComponents: (props: CollectionProps, filepath: string) => {
+      switch (props._collection) {
+        case COLLECTIONS.AUTHOR:
+          return (
+            <AuthorCollectionCustomReference
+              name={props.name}
+              description={props.description}
+            />
+          )
+
+        case COLLECTIONS.POST:
+          return <PostCollectionCustomReference title={props.title} />
+
+        default:
+          return filepath
+      }
+    },
+  },
+  collections: ['author', 'post'],
+}
+```
+
+In this example, the Author collection and Post collection are used as a reference for the Page collection. The `selectComponents`  function allows you to create a custom reference display by providing two parameters: `props` and `filepath`.
+
+* `props` represents the fields from the reference collection (in this case, the fields from the Author collection).
+* `filepath` represents the file path of the referenced document within the Author collection.
+* AuthorCollectionCustomReference and PostCollectionCustomReference are the custom components that I want to display in the reference select dropdown.
+
+```ts
+// You should define the types for the fields in the author collection for typesafety
+export interface AuthorProps {
+  name: string
+  description: string
+  _collection: 'author'
+}
+
+// You should define the types for the fields in the post collection for typesafety
+export interface PostProps {
+  title: string
+  excerpt: string
+  _collection: 'post'
+}
+
+// You should define the list of collection used in the reference for typesafety
+export enum COLLECTIONS {
+  AUTHOR = 'author',
+  POST = 'post',
+}
+
+export type CollectionProps = AuthorProps | PostProps
+
+```
+
+**Figure : Recommended Implementation for Custom Reference Selection with Type Safety.**
+
+```ts
+{
+      label: 'Authors',
+      name: 'author',
+      path: 'content/author',
+      format: 'md',
+      fields: [
+        {
+          type: 'string',
+          label: 'Name',
+          name: 'name',
+          required: true,
+        },
+        {
+          type: 'string',
+          label: 'description',
+          name: 'description',
+          required: true,
+        },
+      ],
+    },
+```
+
+**Figure: Author Collection Schema**
+
+You can enforce type safety in your custom reference selection, reducing the likelihood of errors and ensuring consistency across your application.
+
+Using the AuthorProps interface as an example, the name and description fields are predefined in the schema and will be passed to the selectComponents function. The \_collection field is automatically provided by the system to represent the specific collection (author in this case).
+
+![](https://res.cloudinary.com/forestry-demo/image/upload/v1724068658/reference_field/custom-reference-field-ui_ojtrlx.png)Figure: UI for custom reference select
```

Comment:
As opposed to providing figures, in the rest of the docs we add something like 
"See the example below:" or similar in the proceeding paragraph. (just for consistency).

Reply:
Hmmm, not quite get what you mean, i made new changes ,you can review it again